Index: u-boot-2022.01/board/maxlinear/lightning/lightning.c
===================================================================
--- u-boot-2022.01.orig/board/maxlinear/lightning/lightning.c
+++ u-boot-2022.01/board/maxlinear/lightning/lightning.c
@@ -25,6 +25,9 @@
 #ifdef CONFIG_MXL_OF_DTB_UBOOT_EXT
 #include <asm/sections.h>
 #endif /* CONFIG_MXL_OF_DTB_UBOOT_EXT */
+#include <net.h>
+#include <stdlib.h>
+#include <prpl/mfg.h>
 
 #define MALLOC_BASE_PRE_UBOOT	0x00700000
 #define UBOOT_SRAM_OFFS_LOC		CONFIG_SPL_TEXT_BASE + CONFIG_SPL_PAD_SIZE
@@ -820,6 +823,27 @@ int board_late_init(void)
 	if (CONFIG_IS_ENABLED(LED))
 		led_default_state();
 
+	/* Set MAC address from SMD */
+	if (IS_ENABLED(CONFIG_PRPL_MFG)) {
+		const mfg_entry_t *mfg_entry;
+		char macaddr[32];
+
+		if (prpl_mfg_init())
+			printf("%s: prpl_mfg_init failed.\n", __func__);
+
+		mfg_entry = prpl_mfg_get_entry_by_name("BASE_MAC_ADDRESS");
+		if (!mfg_entry || !mfg_entry->data) {
+			printf("%s: can't find BASE_MAC_ADDRESS in MFG\n", __func__);
+		} else {
+			sprintf(macaddr, "%pM", mfg_entry->data);
+			if (!is_valid_ethaddr(macaddr)) {
+				printf("%s: invalid mac address (%s)\n", __func__, macaddr);
+			} else {
+				env_set("ethaddr", macaddr);
+				printf("Set ethaddr to %s\n", macaddr);
+			}
+		}
+	}
 	return 0;
 }
 
