diff --git a/lib/prpl/mfg.c b/lib/prpl/mfg.c
index f167ce57c..19dc1bea7 100644
--- a/lib/prpl/mfg.c
+++ b/lib/prpl/mfg.c
@@ -106,6 +106,7 @@ static void *prpl_mfg_find_dtb(const char *mfg_part_name)
 	if (board_read_firmware(mfg_part_name, mfg_data) < 0) {
 		log_err("ERROR: unable to read data from flash\n");
 		free(mfg_data);
+		mfg_data = NULL;
 		goto out;
 	}
 	if (IS_ENABLED(CONFIG_PRPL_MFG_GLOBAL_SIG_CHECK)) {
@@ -115,12 +116,14 @@ static void *prpl_mfg_find_dtb(const char *mfg_part_name)
 		if (ret) {
 			log_err("ERROR: Failed to verify global header signature of mfg data\n");
 			free(mfg_data);
+			mfg_data = NULL;
 			env_set("pre_load_sig_info_path", NULL);
 			goto out;
 		}
 		if (fdt_magic(mfg_data + image_load_offset) != FDT_MAGIC) {
 			log_err("ERROR: Failed to retrieve the mfg data from flash\n");
 			free(mfg_data);
+			mfg_data = NULL;
 			goto out;
 		}
 		memmove(mfg_data, mfg_data + image_load_offset, fdt_totalsize(mfg_data + image_load_offset));
@@ -128,6 +131,7 @@ static void *prpl_mfg_find_dtb(const char *mfg_part_name)
 		if (fdt_magic(mfg_data) != FDT_MAGIC) {
 			log_err("ERROR: Failed to retrieve the mfg data from flash\n");
 			free(mfg_data);
+			mfg_data = NULL;
 			goto out;
 		}
 	}
