diff --git a/arch/x86/lib/bootm.c b/arch/x86/lib/bootm.c
index 55265094c..0c0d974cb 100644
--- a/arch/x86/lib/bootm.c
+++ b/arch/x86/lib/bootm.c
@@ -69,6 +69,101 @@ int arch_fixup_memory_node(void *blob)
 }
 #endif
 
+#if defined(CONFIG_VENDOR_MAXLINEAR) && defined(CONFIG_CMD_ICC_OVERLAY)
+void read_cma_size(void) {
+	const void *fdt;
+	int node_offset, len, bootargs_len;
+	const char *prop;
+	char *str_rootfs_loadaddr = NULL;
+	char *str_rootfs_alignment = NULL;
+	char *str_bootargs = NULL;
+	char *str_new_bootargs = NULL;
+	u32 rootfs_load_addr = 0;
+	ulong cma_size, cma_alignment, cma_start, rootfs_alignment;
+	ulong dtb_loadaddr = 0x8000000;
+
+	fdt = (void *)dtb_loadaddr;
+
+	str_bootargs = env_get("bootargs");
+
+	if (!str_bootargs) {
+		printf("[%s] bootargs not found\n", __func__);
+		return;
+	}
+
+	bootargs_len = strlen(str_bootargs) + 30;
+	str_new_bootargs = malloc(bootargs_len);
+
+	if (!str_new_bootargs) {
+		printf("[%s] Failed to allocate memory\n", __func__);
+		return;
+	}
+
+	node_offset = fdt_path_offset(fdt, "/reserved-memory/linux,cma");
+
+	if (node_offset < 0) {
+		printf("[%s] Error: Cannot find /reserved-memory/linux,cma node, error code: %d\n", __func__, node_offset);
+		return;
+	}
+
+	// Get the CMA size
+	prop = fdt_getprop(fdt, node_offset, "size", &len);
+
+	if (!prop) {
+		printf("[%s] Error: Cannot find size property\n", __func__);
+		return;
+	}
+
+	cma_size = fdt32_to_cpu(*(const uint32_t *)prop);
+
+	// Get the CMA alignment
+	prop = fdt_getprop(fdt, node_offset, "alignment", &len);
+
+	if (!prop) {
+		printf("[%s] Error: Cannot find alignment property\n", __func__);
+		return;
+	}
+
+	cma_alignment = fdt32_to_cpu(*(const uint32_t *)prop);
+
+	str_rootfs_loadaddr = env_get("rootfs_loadaddr");
+	str_rootfs_alignment = env_get("rootfs_alignment");
+
+	if ((str_rootfs_loadaddr != NULL) && (str_rootfs_alignment != NULL)) {
+		rootfs_load_addr = simple_strtoul(str_rootfs_loadaddr, NULL, 16);
+		rootfs_alignment = simple_strtoul(str_rootfs_alignment, NULL, 16);
+
+		if ((rootfs_load_addr != 0) && (rootfs_alignment != 0)) {
+			debug("[%s] rootfs_loadaddr = 0x%08X\n", __func__, rootfs_load_addr);
+			debug("[%s] rootfs_alignment = 0x%lx\n", __func__, rootfs_alignment);
+
+			cma_start = rootfs_load_addr - cma_size;
+			snprintf(str_new_bootargs, bootargs_len, "%s cma=0x%lx@0x%lx", str_bootargs, cma_size, cma_start);
+
+			if (env_set("bootargs", str_new_bootargs)) {
+				printf("[%s] Failed to set bootargs\n", __func__);
+				free(str_new_bootargs);
+				return;
+			}
+			//cma check
+			if (cma_size % cma_alignment != 0) {
+				printf("[%s] Error: CMA size is not aligned, pls check DTS\n", __func__);
+				free(str_new_bootargs);
+				return;
+			} else if (rootfs_alignment % cma_alignment != 0) {
+				printf("[%s] Error: CMA size is not a multiple or larger than RootFS size\n", __func__);
+				free(str_new_bootargs);
+				return;
+			}
+		} else
+			debug("[%s] no value for rootfs_loadaddr or rootfs_alignment\n", __func__);
+	} else
+		debug("[%s] not found var rootfs_loadaddr or rootfs_alignment\n", __func__);
+
+	free(str_new_bootargs);
+}
+#endif
+
 /* Subcommand: PREP */
 static int boot_prep_linux(bootm_headers_t *images)
 {
@@ -160,6 +255,10 @@ static int boot_prep_linux(bootm_headers_t *images)
 		goto error;
 	}
 
+#ifdef CONFIG_CMD_ICC_OVERLAY
+	read_cma_size();
+#endif
+
 	printf("Setup at %#08lx\n", images->ep);
 	ret = setup_zimage((void *)images->ep, cmd_line_dest,
 			0, images->rd_start,
