diff --git a/arch/x86/lib/bootm.c b/arch/x86/lib/bootm.c
index 55265094c..5b3cee76e 100644
--- a/arch/x86/lib/bootm.c
+++ b/arch/x86/lib/bootm.c
@@ -238,6 +238,11 @@ static int boot_jump_linux(bootm_headers_t *images)
 	printf("## Transferring control to Linux (at address %08lx, kernel %08lx) ...\n",
 	      images->ep, images->os.load);
 
+#ifdef CONFIG_CMD_ICC_OVERLAY
+	extern void icc_sync_cmd(void);
+	icc_sync_cmd();
+#endif
+
 #ifdef CONFIG_CMD_ICC
 	extern void icc_uboot_bootm(void);
 
diff --git a/board/maxlinear/Kconfig b/board/maxlinear/Kconfig
index 670834ce9..675d1ac67 100644
--- a/board/maxlinear/Kconfig
+++ b/board/maxlinear/Kconfig
@@ -159,15 +159,15 @@ config MAX_BOOT_COUNTER
 	default 3
 
 config LOADADDR
-	hex
+	hex "System load address"
 	default 0x8200000
 
 config DTB_LOADADDR
-	hex
+	hex "Device tree load address"
 	default 0x8000000
 
 config ROOTFS_LOADADDR
-	hex
+	hex "Root-filesystem load address"
 	default 0x40000000
 
 config BOOTCOMMAND
diff --git a/board/maxlinear/lightning/icc/drv_icc.c b/board/maxlinear/lightning/icc/drv_icc.c
index 0b5793cc8..edc43adca 100644
--- a/board/maxlinear/lightning/icc/drv_icc.c
+++ b/board/maxlinear/lightning/icc/drv_icc.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright © 2022 MaxLinear, Inc.
+ * Copyright ï¿½ 2022 MaxLinear, Inc.
  *
  */
 #include <common.h>
@@ -285,3 +285,38 @@ void icc_uboot_bootm(void)
 
 	icc_uboot(CMD_ICC_CFG_FIREWALL, kernel_start_addr, kernel_ROend_addr, 0, 0);
 }
+
+/* this function will be called in boot_jump_linux() in arch/x86/lib/bootm.c */
+void icc_sync_cmd(void)
+{
+	char *str_md_ro_start = NULL;
+	char *str_md_ro_end   = NULL;
+	char *str_rootfs_loadaddr = NULL;
+	u32 kernel_start_addr = 0;
+	u32 kernel_ROend_addr = 0;
+	u32 rootfs_load_addr = 0;
+
+	str_md_ro_start = env_get("md_ro_start");
+	str_md_ro_end   = env_get("md_ro_end");
+	str_rootfs_loadaddr = env_get("rootfs_loadaddr");
+
+	if ((str_md_ro_start != NULL) &&
+	    (str_md_ro_end   != NULL) &&
+	    (str_rootfs_loadaddr != NULL)) {
+		kernel_start_addr = simple_strtoul(str_md_ro_start, NULL, 16);
+		kernel_ROend_addr = simple_strtoul(str_md_ro_end, NULL, 16);
+		rootfs_load_addr = simple_strtoul(str_rootfs_loadaddr, NULL, 16);
+
+		if ((kernel_start_addr != 0) &&
+		    (kernel_ROend_addr != 0) &&
+		    (rootfs_load_addr != 0)) {
+			debug("[%s] kernel_start_addr = 0x%08X\n", __func__, kernel_start_addr);
+			debug("[%s] kernel_ROend_addr = 0x%08X\n", __func__, kernel_ROend_addr);
+			debug("[%s] rootfs_loadaddr = 0x%08X\n", __func__, rootfs_load_addr);
+
+			icc_uboot(1, kernel_start_addr, kernel_ROend_addr, rootfs_load_addr, CONFIG_LGM_MEM_SIZE);
+		} else
+			printf("[%s] no value for md_ro_start, md_ro_end and rootfs_loadaddr\n", __func__);
+	} else
+		debug("[%s] not found var md_ro_start, md_ro_end and rootfs_loadaddr\n", __func__);
+}
\ No newline at end of file
diff --git a/cmd/Kconfig b/cmd/Kconfig
index f80bf4760..ab1b99a1c 100644
--- a/cmd/Kconfig
+++ b/cmd/Kconfig
@@ -2172,6 +2172,12 @@ config CMD_ICC
 	help
 	  User application to send message to TEP
 
+config CMD_ICC_OVERLAY
+	bool "LGM inter chip communication for overlay"
+	default n
+	help
+	  User application to send message to TEP for overlay
+
 config CMD_IA_SWITCH
 	bool "switch from IA_BOOT to IA_VMM mode"
 	default n
