diff --git a/board/maxlinear/lightning/lightning.c b/board/maxlinear/lightning/lightning.c
index 06f5d828e..f13187b03 100644
--- a/board/maxlinear/lightning/lightning.c
+++ b/board/maxlinear/lightning/lightning.c
@@ -467,24 +467,44 @@ err:
 static void pre_calculate_rootfs_variables(void)
 {
 	u32 ddr_top, align = 0;
-	u32 rootfs_size, rootfsload_addr, rootfsload_addr_end;
+	u32 rootfs_size = 0, rootfsload_addr, rootfsload_addr_end;
 	u64 mb = 1024 * 1024;
 	u32 aligned_boundary = ROOTFS_SIZE_ALIGN_VAL;
 	u32 boot_partition;
+	char *active_bank = NULL;
 
-	/* This is to support bank switch mechanism */
-	boot_partition = env_get_ulong("boot_p", 16, 0);
-
-	/* Get the filesize of the boot partiton rootfs.
-	 * If for some reason the rootfs size is not found,
-	 * default to a known rootfs size.
+	/* rootfs size alignment follows below logic
+	if active_bank is A, active bank rootfs size (rootfs_a_filesize) will be used
+	if active_bank is B, inactive bank rootfs size (rootfs_b_filesize) will be used
+	if active_bank is not found , boot_p will be used to decide.
+	 if boot_p is also not found, rootfs_size will be set to default size
 	*/
-	if (boot_partition == 0)
-		rootfs_size = env_get_ulong("rootfs_a_filesize", 16, 0);
-	else if (boot_partition== 1)
-		rootfs_size = env_get_ulong("rootfs_b_filesize", 16, 0);
-	else
-		rootfs_size = 0;
+	active_bank = env_get("active_bank");
+	if (active_bank != NULL) {
+		if (active_bank[0] == 'A') {
+			rootfs_size = env_get_ulong("rootfs_a_filesize", 16, 0);
+			debug("rootfs_a_filesize is used %08x\n", rootfs_size);
+		}
+		else if (active_bank[0] == 'B') {
+			rootfs_size = env_get_ulong("rootfs_b_filesize", 16, 0);
+			debug("rootfs_b_filesize is used %08x\n", rootfs_size);
+		}
+	}
+	if ( rootfs_size == 0 ) {
+		/* This is to support bank switch mechanism */
+		boot_partition = env_get_ulong("boot_p", 16, 0);
+
+		/* Get the filesize of the boot partiton rootfs.
+		* If for some reason the rootfs size is not found,
+		* default to a known rootfs size.
+		*/
+		if (boot_partition == 0)
+			rootfs_size = env_get_ulong("rootfs_a_filesize", 16, 0);
+		else if (boot_partition== 1)
+			rootfs_size = env_get_ulong("rootfs_b_filesize", 16, 0);
+		else
+			rootfs_size = 0;
+	}
 
 	if (rootfs_size == 0) {
 		printf("rootfs_a_filesize not found, using default size\n");
