diff --git a/board/maxlinear/lightning/lightning.c b/board/maxlinear/lightning/lightning.c
index 06f5d828e..797438e86 100644
--- a/board/maxlinear/lightning/lightning.c
+++ b/board/maxlinear/lightning/lightning.c
@@ -681,7 +681,7 @@ static void set_kernel_boot_freq(void)
 static int chk_udt(void)
 {
 	int env_changed = 0;
-	u32 chk_udt_stat = env_get_ulong("udt_status", 16, 0);
+	u32 chk_udt_stat = env_get_ulong("udt_status", 10, 0);
 	u32 chk_boot_cnt = env_get_ulong("boot_count", 16, 0);
 	u32 chk_max_boot_cnt = env_get_ulong("max_boot_count", 16, 0);
 	u32 chk_recovery_img = env_get_ulong("recovery_image", 16, 0);
@@ -711,7 +711,7 @@ static int chk_udt(void)
 		}
 	}
 
-	debug("udt_stat: %x | boot_count: %x\n",
+	debug("udt_stat: %u | boot_count: %x\n",
 		chk_udt_stat, chk_boot_cnt);
 
 #if !defined(CONFIG_BOOT_FROM_MEM)
diff --git a/board/maxlinear/lightning/udt.c b/board/maxlinear/lightning/udt.c
index 7ad9a9373..1bf037a8e 100644
--- a/board/maxlinear/lightning/udt.c
+++ b/board/maxlinear/lightning/udt.c
@@ -11,6 +11,6 @@
 
 void update_udt_failure(int status, int failure) 
 {
-	env_set_hex("udt_status", status);	
-	env_set_hex("udt_error", failure);
+	env_set_ulong("udt_status", status);
+	env_set_ulong("udt_error", failure);
 }
